### ISSUE TITLE
Narrow down pokemon identification with candy

### ISSUE THREAD

**Issue by nahojjjen** [farkam135/GoIV#24]
An idea I've discussed a few times on reddit, is to narrow down the possibilities of what pokemon the user is looking at by watching candy names.

By doing this, you can narrow down the pokemon-picker to a single evolution line.

Example: A user has renamed a weedle to "snores", and the app erroneously recognizes it as a snorlax. If the app instead read that the pokemon wants to eat "weedle candy", then the options of possible pokemon are instead weedle, kakuna and beedrill. The app could then instead pick whichever of those 3 alternatives are closest to "snores", and when the user then corrects the input with the dropdown menu, the menu should only contain the 3 possible options, instead of all 151 in alphabetical order.

What would need to be done? My suggestions:
The application would need to ocr the candy name. If we in the Pokemon object allow another field which stores the "next evolution", some logic could solve the rest.

For example
        pokemon.add(new Pokemon("pokemon name", "evolution name", 110, 76, 50));

So an algorithm that figures out an evolution chain could simply check the candy name, for example bulbasaur, and then see if that pokemon has a non-null evolution name.

**Comment by agrajaghh** [comment]
what is with OCR errors? How to correct them without displaying the full 151?

**Comment by nahojjjen** [comment]
Reopened because it's still not narrowing down the dropdown list

**Comment by Eccenux** [comment]
I see there were some changes in `showInfoLayout`. It was better now its worse ;-).

If you changed the name of your pokemon this will not give you expected result:
```java
            if (possiblePoke[1] < pokeCalculator.get(possibleCandy[0]).getDistance(origPokemonName)) {
                pokemonList.setSelection(possiblePoke[0]);
            } else {
                pokemonList.setSelection(possibleCandy[0]);
            }
```
`possiblePoke` is best match to any pokemon in pokedex
`possibleCandy` is best match of candy name in pokedex 

For pokemon named "Blah", candy "Mr. Mime"
`possiblePoke` would be nonsens.
`possibleCandy` would be an actual match.

`possibleCandy` is good for all non-evolved pokemons. It might not be good for evolved pokemon, but still probably better if `possiblePoke[1]` (distance to any known pokemon) is high.

So this would be better:
```java
            if (possiblePoke[1] < possibleCandy[1]) {
                pokemonList.setSelection(possiblePoke[0]);
            } else {
                pokemonList.setSelection(possibleCandy[0]);
            }
```

**Comment by Eccenux** [comment]
BTW. I think if pokemon list would be sorted by index (not name) then above would probably just work.

**Comment by Blaisorblade** [comment]
>  It was better now its worse ;-).

> If you changed the name of your pokemon this will not give you expected result:

Hi, I did that change. Rationale: I rename my Pokemons with a suffix (IV), and my Vaporeon89 were recognized as Eevee. That's why I specifically avoided `possiblePoke[1] < possibleCandy[1]`.

> I think if pokemon list would be sorted by index (not name) then above would probably just work.

Same idea, opened #130. Or we could narrow the list as suggested by @nahojjjen if candy OCR is always reliable, or in some mixture—but sorting by index would be better for me (it's bad if you don't know the indexes but I think one knows it enough).

Even better, we could order the list by likelihood, with the guess from the candy and its evolutions close to the top, but the name-based one before (IMHO).

One could maybe also have a setting named "Trust Pokemon names" but I guess it'd be nice to get something decent without that setting.

**Comment by nahojjjen** [comment]
closed, discussion in #130
